toolkit/tools/scheduler/schedulerutils/pkglicensechecker.go

```diff
@@ -0,0 +1,179 @@
+	if !licensecheck.IsValidLicenseCheckMode(config.Mode) {
+		err = fmt.Errorf("invalid license check mode: %s", config.Mode)
+		return nil, err
+	}
+
+	if config.Mode == licensecheck.LicenseCheckModeNone {
+		return &PackageLicenseChecker{checkerMode: licensecheck.LicenseCheckModeNone}, nil
+	}
+
+	rpmCopyDir := filepath.Join(config.BuildDirPath, "rpms_to_scan")
+	err = directory.EnsureDirExists(rpmCopyDir)
+	if err != nil {
+		err = fmt.Errorf("failed to create directory to stage RPMs:\n%w", err)
+		return nil, err
+	}
+
+	return &PackageLicenseChecker{
+		// Lazy initialize the license checker chroot. We can wait until a build is actually done, and has RPMS to scan,
+		// before starting it. If we start it now, it will block us starting any graph processing until the chroot is
+		// ready.
+		licenseChecker: nil,
+		config:         config,
+
+		rpmDirPath:  rpmCopyDir,
+		checkerMode: config.Mode,
+		resultsFile: config.ResultsFile,
+		summaryFile: config.SummaryFile,
+	}, nil
+}
+
+// Initialize the license checker chroot if needed.
+func (p *PackageLicenseChecker) startLicenseCheckerChroot() (err error) {
+	if p.licenseChecker != nil {
+		return nil
+	}
+	p.licenseChecker, err = licensecheck.New(p.config.BuildDirPath,
+		p.config.WorkerTarPath, p.rpmDirPath, p.config.NameFilePath, p.config.ExceptionFilePath, p.config.DistTag)
+	if err != nil {
+		err = fmt.Errorf("failed to create license checker for use with packages:\n%w", err)
+		return err
+	}
+	return nil
+}
+
+func (p *PackageLicenseChecker) CleanupPackageLicenseChecker() error {
+	if p.licenseChecker != nil {
+		return p.licenseChecker.Cleanup()
+	} else {
+		return nil
+	}
+}
+
+// CheckPkgLicenses will check each RPM passed in for license issues by copying them to a dedicated folder and running
+// the license checker. This function will initialize the license checker if it has not already been initialized.
+// Each call will accumulate the results of the license checks in the license checker instance. After all RPMs have been
+// checked, the results can be saved via SaveResults().
+func (p *PackageLicenseChecker) CheckPkgLicenses(rpmPaths []string) (hasWarning, hasError bool, err error) {
+	if len(rpmPaths) == 0 || p.checkerMode == licensecheck.LicenseCheckModeNone {
+		return false, false, nil
+	}
+
+	if p.licenseChecker == nil {
+		err = p.startLicenseCheckerChroot()
+		if err != nil {
+			err = fmt.Errorf("failed to start license checker chroot:\n%w", err)
+			return false, false, err
+		}
+	}
+
+	// Clear the RPM directory before copying the new RPMs
+	err = file.RemoveDirectoryContents(p.rpmDirPath)
+	if err != nil {
+		err = fmt.Errorf("failed to clear RPM directory before scan:\n%w", err)
+		return false, false, err
+	}
+
+	// Copy the RPMs into the check directory
+	for _, rpmPath := range rpmPaths {
+		err = file.Copy(rpmPath, filepath.Join(p.rpmDirPath, filepath.Base(rpmPath)))
```

Comment:
Use symlink to avoid an actual copy?

Reply:
Needed to use a hardlink, the symlink breaks when you enter the chroot.